bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -97,96 +99,151 @@ protected NAWelcomeHome updateReadings() {
+        }
+
+        if (NAWebhookCameraEvent.EventTypeEnum.MOVEMENT.toString().equals(event.getType())) {
+            detectedObjectTypes.add(NAWebhookCameraEvent.EventTypeEnum.MOVEMENT.name());
+        }
+
+        event.getEventList().forEach(subEvent -> {
+            String detectedObjectType = subEvent.getType().name();
+            detectedObjectTypes.add(detectedObjectType);
+        });
+        return detectedObjectTypes;
+    }
+
+    private Optional<String> findEventMessage() {
+        if (lastEvent.isPresent()) {
+            @Nullable String message = lastEvent.get().getMessage();
+            if (message != null) {
+                return Optional.of(message);
+            }
+
+            return findFirstSubEvent(lastEvent).map(NAWelcomeSubEvent::getMessage);
+        }
+        return Optional.empty();
+    }
+
     /**
      * Returns the Url of the picture
      *
      * @return Url of the picture or null
      */
-    protected String getSnapshotURL(NAWelcomeSnapshot snapshot) {
-        String result = null;
-
-        if (snapshot != null && snapshot.getId() != null && snapshot.getKey() != null) {
-            result = WELCOME_PICTURE_URL + "?" + WELCOME_PICTURE_IMAGEID + "=" + snapshot.getId() + "&"
-                    + WELCOME_PICTURE_KEY + "=" + snapshot.getKey();
-        } else {
-            logger.debug("Unable to build snapshot url for Home : {}", getId());
+    protected Optional<String> findSnapshotURL() {
+        if (lastEvent.isPresent()) {
+            @Nullable NAWelcomeSnapshot snapshot = lastEvent.get().getSnapshot();
+            if (snapshot == null) {
+                snapshot = findFirstSubEvent(lastEvent).map(NAWelcomeSubEvent::getSnapshot).orElse(null);
+            }
+
+            if (snapshot != null && snapshot.getId() != null && snapshot.getKey() != null) {
+                return Optional.of(WELCOME_PICTURE_URL + "?" + WELCOME_PICTURE_IMAGEID + "=" + snapshot.getId() + "&"
+                        + WELCOME_PICTURE_KEY + "=" + snapshot.getKey());
+            } else {
+                logger.debug("Unable to build snapshot url for Home : {}", getId());
+            }
         }
-        return result;
+        return Optional.empty();
     }
 
     @Override
     protected @Nullable Integer getDataTimestamp() {
         return dataTimeStamp;
     }
+
+    private State getPlaceInfo(Function<NAWelcomePlace, String> infoGetFunction) {
+        return Optional.ofNullable(device).map(
+                d -> toStringType(infoGetFunction.apply(d.getPlace()))
+        ).orElse(UnDefType.UNDEF);
+    }
+
+    private static Optional<NAWelcomeSubEvent> findFirstSubEvent(Optional<NAWelcomeEvent> event) {
+        if (event.isPresent()) {
+            List<NAWelcomeSubEvent> subEvents = event.get().getEventList();
+            if (subEvents != null && !subEvents.isEmpty()) {
+                return Optional.of(subEvents.get(0));
+            }
+        }
+        return Optional.empty();
```

Comment:
I think this should work.
```suggestion
        return event.map(NAWelcomeEvent::getEventList).flatMap(subEvents -> subEvents.stream().findFirst());
```

Reply:
done